packages/delivery-electron/delivery.js

```diff
@@ -13,6 +13,14 @@ const delivery = (client, filestore, net, app) => {
 
     const req = net.request(opts, response => {
       req.removeListener('error', errorHandler)
+
+      // in some cases (e.g. proxy authentication errors) Electron net
+      // will emit an error event on the response stream
+      response.on('error', (err) => {
+        err.isRetryable = isRetryable(response.statusCode)
+        cb(err)
```

Comment:
Will this not call `cb` twice? Once on line 25 & once here

Reply:
Well yes, actually it's the `cb` call on line 34 because the response isn't successful, but it's even worse than that! 

It looks like Electron actually emits errors on both the response _and_ the request (as well as returning the response) - so just handling the response error isn't enough. 

We do have an error handler on the request, but this is removed at the start of the response callback on line 15, presumably to prevent the request error handler being called if there's a response. 

The problem is that without that error handler, the request error will also fall through to the uncaught exception handler, and so we still end up in the same infinite request loop even if we handle the response error.

I think we'll need to remove line 15 and keep both of these error handlers - maybe we should have some kind of `errorHandled` flag which we set before calling `cb`, so that a failed request is only handled once by one of these 3 handlers?

Reply:
Updated - if the response error has already been handled any subsequent errors on the request/response will be ignored